txredisapi.py

```diff
@@ -2380,14 +2380,17 @@ def __init__(self, isLazy=False, handler=ConnectionHandler):
 
 
 def makeConnection(host, port, dbid, poolsize, reconnect, isLazy,
-                   charset, password, connectTimeout, replyTimeout,
+                   charset, password, use_ssl, connectTimeout, replyTimeout,
```

Comment:
I suggest pass here `ssl_context_factory` argument of type `ClientContextFactory` with default value of `None` instead of just a bool. Some users may have specific requirements for certificate validation process so they may want to pass custom ssl context instead of the default one. If `None` is passed, no ssl will be used.

We may go further and treat this argument as `Optional[Union[ClientContextFactory, bool]]` with a following meaning:
* `None` — no SSL
* `False` — no SSL
* `True` — SSL is used with default `ssl.ClientContextFactory()` instance
* `ClientContextFactory` instance — SSL with passed context factory instance

This way we won't break compatibility with your original PR if you already using it in production as your may still just pass `True` and have SSL with default context factory. The downside is that `Union`s looks not too pythonic for me...

What do you think?